pkg/cmd/cli/secrets/new.go

```diff
@@ -154,36 +155,50 @@ func (o *CreateSecretOptions) BundleSecret() (*kapi.Secret, error) {
 	secretData := make(map[string][]byte)
 
 	for _, source := range o.Sources {
-		info, err := os.Stat(source)
+		keyName, filePath, err := parseSource(source)
+		if err != nil {
+			return nil, err
+		}
+
+		info, err := os.Stat(filePath)
 		if err != nil {
 			switch err := err.(type) {
 			case *os.PathError:
-				return nil, fmt.Errorf("Error reading %s: %v", source, err.Err)
+				return nil, fmt.Errorf("Error reading %s: %v", filePath, err.Err)
 			default:
-				return nil, fmt.Errorf("Error reading %s: %v", source, err)
+				return nil, fmt.Errorf("Error reading %s: %v", filePath, err)
 			}
 		}
 
 		if info.IsDir() {
-			fileList, err := ioutil.ReadDir(source)
+			if strings.Contains(source, "=") {
+				return nil, errors.New("Cannot give a key name for a directory path.")
+			}
+			fileList, err := ioutil.ReadDir(filePath)
 			if err != nil {
-				return nil, fmt.Errorf("Error listing files in %s: %v", source, err)
+				return nil, fmt.Errorf("Error listing files in %s: %v", filePath, err)
 			}
 
 			for _, item := range fileList {
-				itemPath := path.Join(source, item.Name())
+				itemPath := path.Join(filePath, item.Name())
 				if !item.Mode().IsRegular() {
 					if o.Stderr != nil && o.Quiet != true {
 						fmt.Fprintf(o.Stderr, "Skipping resource %s\n", itemPath)
 					}
 				} else {
-					if err := readFile(itemPath, secretData); err != nil {
+					keyName = path.Base(itemPath)
+					filePath = itemPath
```

Comment:
looks like this will cause wonky things to happen since you're joining with filepath above.  Why do you need this anyway?  I would have expected this to fail a unit test.  Add one that this will fail.